src/models/game/epics.js

```diff
@@ -0,0 +1,235 @@
+        playerHand: newerPlayerHand,
+        aiHand: newerAiHand,
+        changedPlayerHand: Array(5).fill(null),
+        playerBet: newPlayerBet + bet,
+        aiBet: newAiBet + bet,
+        pot: newPot + bet * 2,
+        playerMoney: newPlayerMoney - bet,
+        aiMoney: newAiMoney - bet,
+        phase: newPhase,
+      });
+    })
+  );
+
+const foldEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(fold.type),
+    map(({ payload }) => {
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPhase = 4;
+      const newPot = pot(state$.value);
+
+      return gameFolded({
+        playerBet: 0,
+        aiBet: 0,
+        phase: newPhase,
+        pot: 0,
+        playerMoney: newPlayerMoney,
+        aiMoney: newAiMoney + newPot,
+      });
+    })
+  );
+
+const raiseEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(raise.type),
+    map(({ payload }) => {
+      const newPlayerBet = playerBet(state$.value);
+      const newAiBet = aiBet(state$.value);
+      const bet = 100;
+
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPot = pot(state$.value);
+
+      if (newPlayerMoney > 0) {
+        return betRaised({
+          playerMoney: newPlayerMoney - bet,
+          aiMoney: newAiMoney - bet,
+          pot: newPot + bet * 2,
+          playerBet: newPlayerBet + bet,
+          aiBet: newAiBet + bet,
+        });
+      } else if (newPlayerMoney <= 0) {
+        return betNotRaised({});
+      }
+    })
+  );
+
+const replaceEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(replace.type),
+    map(({ payload }) => {
+      const newPlayerHand = playerHand(state$.value).slice();
+      const newDeck = deck(state$.value).slice();
+      const newChangedPlayerHand = changedPlayerHand(state$.value).slice();
+      const drawnCardIndex = newPlayerHand.indexOf(payload);
+      if (newChangedPlayerHand[drawnCardIndex] === null) {
+        newChangedPlayerHand.splice(drawnCardIndex, 1, newDeck[0]);
+        newPlayerHand.splice(
+          drawnCardIndex,
+          1,
+          newChangedPlayerHand[drawnCardIndex]
+        );
+        newDeck.splice(0, 1);
+        return cardReplaced({
+          deck: newDeck,
+          playerHand: newPlayerHand,
+          changedPlayerHand: newChangedPlayerHand,
+        });
+      } else return cardNotReplaced({});
+    })
+  );
+const checkEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(check.type),
+    map(({ payload }) => {
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPlayerBet = playerBet(state$.value);
+      const newAiBet = aiBet(state$.value);
+      const newPot = pot(state$.value);
+      const newPhase = phase(state$.value);
+      const newPlayerHand = playerHand(state$.value).slice();
+      const newAiHand = aiHand(state$.value).slice();
+
```

Comment:
common properties from the object below can be created here under a const commonPayload and be spread in the objects below and replace only the different properties.

Reply:
Done and done.